_posts/2018/2018-04-15-use-hash-fetch-when-using-params-in-controllers.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'Use Hash#fetch when using Rails params in controllers'
+description: ''
+layout: article
+category: ruby
+image:
+  base: '2018/use-hash-fetch-when-using-params-in-controllers'
+  alt: 'Retriever'
+  source: 'https://unsplash.com/photos/Pzu9f6Nby5w'
+  credit: 'Mitchell Orr'
+---
+
+We often need to access a parameter from the URL that isn't part of the regular rails routes. Something passed after
+
+A ruby `Hash` is typically accessed using square brackets, like `hash[key]`, but you can also use the method `fetch`, like `hash.fetch(key)`.
+
+
+## Instead of…
+
+...accessing required parameters from the `params` hash using the typical `#[]` method...
+
+```ruby
+class ThingController < ApplicationController
+  # ... actions
+
+  private
+
+   def an_important_param
+     params[:important_and_required]
+   end
+end
+```
+
+
+## Use…
+
+...the `#fetch` method to raise an error if the required parameter is missing.
+
+```ruby
+class ThingController < ApplicationController
+  # ... actions
+
+  private
+
+   def an_important_param
+     params.fetch(:important_and_required)
+   end
+end
+```
+
+
+## But why?
+
+If you access an item from the `params` hash directly or have built this sort of private method it implies that the hash contains a required value for the correct functioning of your action.
```

Comment:
This needs rejigging for clarity. You're basically saying: if you're trying to access something within a hash similar to how I've done it above, then what you're looking for is probably very important, right?